engine/swarm/services.md

```diff
@@ -470,7 +471,40 @@ To create a service with access to Docker-managed secrets, use the `--secret`
 flag. For more information, see
 [Manage sensitive strings (secrets) for Docker services](secrets.md)
 
-### Control service scale and placement
+### Control service placement
+
+Swarm services provide a few different ways for you to control scale and
+placement of services on different nodes.
+
+- You can specify whether the service needs to run a specific number of replicas
+  or should run globally on every worker node. See
+  [Replicated or global services](#replicated-or-global-services).
+
+- You can configure the service's
+  [CPU or memory requirements](#reserve-memory-or-cpus-for-a-service), and the
+  service will only run on nodes which can meet those requirements.
+
+- [Placement constraints](#placement-constraints) let you configure the service
+  to run only on nodes with specific (arbitrary) metadata set, and cause the
+  deployment to fail if appropriate nodes do not exist. For instance, you can
+  specify that your service should only run on nodes where an arbitrary label
+  `pci_compliant` is set to `true`.
+
+- [Placement preferences](#placement-preferences) let you apply an arbitrary
+  label with a range of values to each node, and spread your service's tasks
+  across those nodes using an algorithm. Currently, the only supported algorithm
+  is `spread`, which  which tries to place them evenly. For instance, if you
+  label each node with a label `rack` which has a value from 1-10, then specify
+  a placement preference keyed on `rack`, then service tasks are placed as
+  evenly as possible across all nodes with the label `rack`, after taking other
+  placement constraints, placement preferences, and other node-specific
+  limitations into account.
```

Comment:
Also note that the preference is just ranking a node higher than other nodes.
Swarm will still spread to nodes that do not match any existing preference if all nodes with a preference are already scheduled (or unavailable for new tasks that matter) but more replicas were requested.
This is because swarm's spread strategy schedules for high availability and the preference just says "prefer these nodes over others", but HA is still the most important thing... hope that makes sense.

Reply:
An example:

If you have `N` nodes with the `rack` label (and then some others), and `N+1` replicas requested, the `+1` will be scheduled on a node that doesn't already have the service on it if there is one regardless of the `rack` label existing.